MINOR: Document endpoints for connector topic tracking (KIP-558)

kkonstantine opened the pull request:
Update the site documentation to include the endpoints introduced with KIP-558 and a short paragraph on how this feature is used in Connect. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

kkonstantine committed:
MINOR: Document endpoints for connector topic tracking (KIP-558)

ewencp commented on docs/connect.html:
```diff
@@ -576,6 +578,10 @@ <h4><a id="connect_administration" href="#connect_administration">Kafka Connect
     </p>
 
+    <p>
+    Starting with 2.5.0, Kafka Connect is using the <code>status.storage.topic</code> to also store information related to the topics that each connector is using. Connect workers use these per connector topic status updates to respond to requests to the REST endpoint <code>GET /connectors/{name }/topics</code> by returning the set of topic names that a connector is using. A request at any time t ...
```
Stylistic nits, but not a big deal:

* I think `Kafka Connect is using` -> `Kafka Connect uses` sounds more natural
* I would capitalize `Connect workers` -> `Connect Workers` (partly just because `Worker` is a class name), though it seems we don't consistently capitalize anyway...
* Nit that may not even show up rendered: extra ' ' in `{name }`.

kkonstantine replied:
Thanks @ewencp! All accepted :)

ewencp reviewed (approved):
A couple of minor nits, but lgtm generally.

It looks like KIP-558 is landing in 2.5, so I assume request here is to merge to trunk and cherry-pick to 2.5 branch as well?

kkonstantine committed:
MINOR: Address Ewen's comments

kkonstantine committed:
MINOR: Fix white space and wording

kkonstantine commented:
Thanks for the review @ewencp !
I've addressed your comments and fixed the text on the second endpoint. 

Indeed, this refers to a KIP that has been merged to 2.5, so this PR should be merged to `trunk` and backported to the `2.5` branch currently.

tobydrake7 commented on docs/connect.html:
```diff
@@ -576,6 +578,10 @@ <h4><a id="connect_administration" href="#connect_administration">Kafka Connect
     </p>
 
+    <p>
+    Starting with 2.5.0, Kafka Connect uses the <code>status.storage.topic</code> to also store information related to the topics that each connector is using. Connect Workers use these per connector topic status updates to respond to requests to the REST endpoint <code>GET /connectors/{name}/topics</code> by returning the set of topic names that a connector is using. A request at any time to the ...
```
```suggestion
    Starting with 2.5.0, Kafka Connect uses the <code>status.storage.topic</code> to also store information related to the topics that each connector is using. Connect Workers use these per-connector topic status updates to respond to requests to the REST endpoint <code>GET /connectors/{name}/topics</code> by returning the set of topic names that a connector is using. A request to the REST endpoint <code>PUT /connectors/{name}/topics/reset</code> resets the set of active topics for a connector and allows a new set to be populated, based on the connector's latest pattern of topic usage. Upon connector deletion, the set of the connector's active topics is also deleted. Topic tracking is enabled by default but can be disabled by setting <code>topic.tracking.enable=false</code>. If you want to disallow requests to reset the active topics of connectors during runtime, set the Worker property <code>topic.tracking.allow.reset=false</code>.
```

tobydrake7 replied:
just a minor suggested change

kkonstantine replied:
Thanks @tobydrake7 ! Incorporated.

tobydrake7 reviewed (approved):
LGTM 👍 minor suggested changes

kkonstantine committed:
MINOR: Address Toby's comment

ewencp commented:
Thanks, merging to trunk and 2.5
